### PR TITLE
Provider order api

### PR DESCRIPTION
**Pull request by hoangnguyen02081993**
Add two function with this order.
    + Get order with orderstatus , starttime and endtime.
    + Set order status with Orderid, orderstatus, message.

    Add its test

***    Edit Order and Order_history table

    With provider-order-api, need more column to act other thing.
    So, add its column to action. ***

### CODE REVIEW

File: foodcopingserver/foodcopingserver/Controllers/Provider/api/ProviderOrderAPIController.cs [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+{
+    public class ProviderOrderAPIController : ApiController
+    {
+        [Route("api/provider/getorder")]
+        [HttpPost]
+        public IHttpActionResult Show([FromBody] ProviderOrderInfoModel input)
+        {
+            ProviderOrderInfoModel result = new ProviderOrderInfoModel();
+
+            Guid sessionGuid = Guid.Empty;
```
Comment (bxdoan) [inline foodcopingserver/foodcopingserver/Controllers/Provider/api/ProviderOrderAPIController.cs]: why dont combine in 1 condition ?

File: foodcopingserver/foodcopingserver/Controllers/Provider/api/ProviderOrderAPIController.cs [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+                    {
+                        Orders = session.provider.orders.Where(o => o.CreateTime >= StartTime && o.CreateTime <= EndTime && o.orderstatusId == input.OrderStatusId && o.providerId == session.providerid).ToList();
+                        Orders_history = session.provider.order_history.Where(o => o.CreateTime >= StartTime && o.CreateTime <= EndTime && o.orderstatusId == input.OrderStatusId && o.providerId == session.providerid).ToList();
+                    }
+                    foreach (var item in Orders)
+                    {
+                        var order = new ProviderOrder()
+                        {
+                            OrderId = item.id.ToString(),
+                            UserFullName = item.user.FirstName + " " + item.user.LastName,
```
Comment (bxdoan) [inline foodcopingserver/foodcopingserver/Controllers/Provider/api/ProviderOrderAPIController.cs]: This application for VNese people, change to Lastname + " " + Firstname

File: foodcopingserver/foodcopingserver/Controllers/Provider/api/ProviderOrderAPIController.cs [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+                            order.details.Add(detail);
+                        }
+                        result.orders.Add(order);
+                    }
+                    foreach (var item in Orders_history)
+                    {
+                        var order = new ProviderOrder()
+                        {
+                            OrderId = item.id.ToString(),
+                            UserFullName = item.user.FirstName + " " + item.user.LastName,
```
Comment (bxdoan) [inline foodcopingserver/foodcopingserver/Controllers/Provider/api/ProviderOrderAPIController.cs]: same above

File: foodcopingserver/foodcopingserver/Controllers/Provider/api/ProviderOrderAPIController.cs [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+                            PaymentStyleString = item.sys_payment_style.name,
+                            OrderStatusString = item.order_status.description,
+                            TotalAmount = item.TotalAmount,
+                            ProviderMessage = item.ProviderMessage,
+                            CreateDate = item.CreateTime.ToString(" yyyy-MM-dd HH:mm:ss")
+                        };
+
+                        foreach (var item1 in item.order_detail_history)
```
Comment (bxdoan) [inline foodcopingserver/foodcopingserver/Controllers/Provider/api/ProviderOrderAPIController.cs]: No need  to increase the number of variable, change name to "value" or "element" or something like this pls

File: foodcopingserver/foodcopingserver/Controllers/Provider/api/ProviderOrderAPIController.cs [hunk 5]
```diff
@@ -0,0 +1,204 @@
...
+                        {
+                            var detail = new ProviderOrderDetail()
+                            {
+                                MenuName = item1.provider_menu.name,
+                                Quantity = item1.Quantity,
+                                SubtotalAmount = item1.SubtotalAmount
```
Comment (bxdoan) [inline foodcopingserver/foodcopingserver/Controllers/Provider/api/ProviderOrderAPIController.cs]: "SubtotalAmount" WTF name of variable? => change to "subTotalAmount" or "SubTotalAmount" and in DB also

File: foodcopingserver/foodcopingserver/Controllers/Provider/api/ProviderOrderAPIController.cs [hunk 6]
```diff
@@ -0,0 +1,204 @@
...
+            {
+                result.result = 1;
+                result.orders = null;
+            }
+
+            return Ok(result);
+        }
+        [Route("api/provider/setorder")]
+        [HttpPost]
+        public IHttpActionResult setOrderStatus([FromBody] ProviderOrderInfoModel input)
```
Comment (bxdoan) [inline foodcopingserver/foodcopingserver/Controllers/Provider/api/ProviderOrderAPIController.cs]: "setOrderStatus" => "SetOrderStatus"

File: foodcopingserver/foodcopingserver/Controllers/Provider/api/ProviderOrderAPIController.cs [hunk 7]
```diff
@@ -0,0 +1,204 @@
...
+        {
+            ProviderOrderInfoModel result = new ProviderOrderInfoModel();
+
+            Guid sessionGuid = Guid.Empty;
+            Guid orderGuid = Guid.Empty;
+            if (input == null || input.sessionId == null || !Guid.TryParse(input.sessionId, out sessionGuid) ||
```
Comment (bxdoan) [inline foodcopingserver/foodcopingserver/Controllers/Provider/api/ProviderOrderAPIController.cs]: combine this into 1 condition

File: foodcopingserver/foodcopingserver/Controllers/Provider/api/ProviderOrderAPIController.cs [hunk 8]
```diff
@@ -0,0 +1,204 @@
...
+                return BadRequest();
+            }
+
+            var session = DataWrapper.FoocopingContext.provider_login_session.Where(s => s.id == sessionGuid).FirstOrDefault();
+
+            if (session != null)
+            {
+                try
+                {
+                    var Order = session.provider.orders.Where(o => o.id == orderGuid).FirstOrDefault();
```
Comment (bxdoan) [inline foodcopingserver/foodcopingserver/Controllers/Provider/api/ProviderOrderAPIController.cs]: "orderGuid" what does it mean? change to "orderGuide" and in DB also

File: foodcopingserver/foodcopingserver/Models/foodcopingDB.Context.cs [hunk 9]
```diff
@@ -76,12 +72,16 @@ protected override void OnModelCreating(DbModelBuilder modelBuilder)
 
     public virtual DbSet<provider_security> provider_security { get; set; }
 
-    public virtual DbSet<provider> providers { get; set; }
-
     public virtual DbSet<provider_error_place> provider_error_place { get; set; }
 
     public virtual DbSet<provider_place> provider_place { get; set; }
 
+    public virtual DbSet<order> orders { get; set; }
```
Comment (bxdoan) [inline foodcopingserver/foodcopingserver/Models/foodcopingDB.Context.cs]: This file is changed NOTHING !!!

File: foodcopingserver/foodcopingserver/Models/order.cs [hunk 10]
```diff
@@ -1,36 +1,62 @@
...
+
+    public virtual sys_payment_style sys_payment_style { get; set; }
+
+    public virtual user user { get; set; }
+
+    public virtual provider provider { get; set; }
+
+}
+
+}
```
Comment (bxdoan) [inline foodcopingserver/foodcopingserver/Models/order.cs]: This file is changed NOTHING !!! What are you doing ?

File: foodcopingserver/foodcopingserver/Models/order_history.cs [hunk 11]
```diff
@@ -1,36 +1,62 @@
...
+    public virtual order_status order_status { get; set; }
+
+    public virtual sys_payment_style sys_payment_style { get; set; }
+
+    public virtual user user { get; set; }
+
+    public virtual provider provider { get; set; }
+
+}
+
```
Comment (bxdoan) [inline foodcopingserver/foodcopingserver/Models/order_history.cs]: This file is changed NOTHING !!! What are you doing ?

File: foodcopingserver/foodcopingserver/Views/ProviderOrderAPITest/Index.cshtml [hunk 12]
```diff
@@ -0,0 +1,197 @@
+﻿<html>
+<head>
+    <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.2.1/jquery.min.js"></script>
```
Comment (bxdoan) [inline foodcopingserver/foodcopingserver/Views/ProviderOrderAPITest/Index.cshtml]: refer to ~/Script/...

File: foodcopingserver/foodcopingserver/Controllers/Provider/api/ProviderOrderAPIController.cs (hunk 1 above)
Comment (hoangnguyen02081993) [reply foodcopingserver/foodcopingserver/Controllers/Provider/api/ProviderOrderAPIController.cs]: I think It's clearly for another reader.
1/ Checking input have or not required variables.
2/ Checking input is in range of its.

File: foodcopingserver/foodcopingserver/Controllers/Provider/api/ProviderOrderAPIController.cs (hunk 2 above)
Comment (hoangnguyen02081993) [reply foodcopingserver/foodcopingserver/Controllers/Provider/api/ProviderOrderAPIController.cs]: Ok, thanks

File: foodcopingserver/foodcopingserver/Controllers/Provider/api/ProviderOrderAPIController.cs (hunk 8 above)
Comment (hoangnguyen02081993) [reply foodcopingserver/foodcopingserver/Controllers/Provider/api/ProviderOrderAPIController.cs]: It's exactly Guid. It's a Guid variable.

File: foodcopingserver/foodcopingserver/Models/order.cs (hunk 10 above)
Comment (hoangnguyen02081993) [reply foodcopingserver/foodcopingserver/Models/order.cs]: It's added two column ( CreateDate and providerId).

File: foodcopingserver/foodcopingserver/Models/order_history.cs (hunk 11 above)
Comment (hoangnguyen02081993) [reply foodcopingserver/foodcopingserver/Models/order_history.cs]: same above

File: foodcopingserver/foodcopingserver/Views/ProviderOrderAPITest/Index.cshtml (hunk 12 above)
Comment (hoangnguyen02081993) [reply foodcopingserver/foodcopingserver/Views/ProviderOrderAPITest/Index.cshtml]: Ok, thanks

**Pull request merged**